src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -368,153 +373,151 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm, bool IsPac
                     query.Boost = isWildcard == false ?
-                        LuceneConfig.SearchCategoryWeight : LuceneConfig.WildcardsSearchCategoryWeight;
+                        LuceneConfig.SearchCategoryWeight : (LuceneConfig.WildcardsSearchCategoryWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.FullCategoryName):
-                    query.Boost = isWildcard == false?
-                        LuceneConfig.SearchCategoryWeight : LuceneConfig.WildcardsSearchCategoryWeight;
+                    query.Boost = isWildcard == false ?
+                        LuceneConfig.SearchCategoryWeight : (LuceneConfig.WildcardsSearchCategoryWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.CategorySplitted):
                     //Under this case the CategorySplitted field will have more weight than NameSplitted
                     query.Boost = isWildcard == false ?
-                        LuceneConfig.SearchNameWeight : LuceneConfig.WildcardsSearchNameWeight;
+                        LuceneConfig.SearchNameWeight : (LuceneConfig.WildcardsSearchNameWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.Description):
                     query.Boost = isWildcard == false ?
-                        LuceneConfig.SearchDescriptionWeight : LuceneConfig.WildcardsSearchDescriptionWeight;
+                        LuceneConfig.SearchDescriptionWeight : (LuceneConfig.WildcardsSearchDescriptionWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.SearchKeywords):
                     query.Boost = isWildcard == false ?
-                       LuceneConfig.SearchTagsWeight : LuceneConfig.WildcardsSearchTagsWeight;
+                       LuceneConfig.SearchTagsWeight : (LuceneConfig.WildcardsSearchTagsWeight - boostOffset);
+                    break;
+                case nameof(LuceneConfig.NodeFieldsEnum.Parameters):
+                    query.Boost = 1;
                     break;
                 default:
                     query.Boost = isWildcard == false ?
-                       LuceneConfig.SearchMetaFieldsWeight : LuceneConfig.WildcardsSearchMetaFieldsWeight;
+                       LuceneConfig.SearchMetaFieldsWeight : (LuceneConfig.WildcardsSearchMetaFieldsWeight - boostOffset);
                     break;
             }
+
+            //This section only applies if the SearchTerm contain empty spaces then is splitted by empty space and try to calculate the weight for each part
+            if (fieldName.Equals(nameof(LuceneConfig.NodeFieldsEnum.Name)) && termSplit == true)
+            {
+                query.Boost = LuceneConfig.WildcardsSearchNameParsedWeight;
+            }
+            else if (termSplit == true)
+            {
+                query.Boost = LuceneConfig.FuzzySearchWeight;
+            }
```

Comment:
Can this block be placed higher above in this method? Since it only depends on `fieldName` and `termSplit`

Reply:
This piece of code is executed when the user for example search for "list join".
We split the text in "list" and "join" a then we try to create the Wildcardquery and set the Weight for each word but due that I tried to unify setting the Weight in the CalculateFieldWeight method so this code needs to be in this place (after the switch) otherwise I will need to modify the next piece of code adding more` if .. else ...`validations:

```
 switch (fieldName)
 {
     case nameof(LuceneConfig.NodeFieldsEnum.Name):
         query.Boost = isWildcard == false ?
             LuceneConfig.SearchNameWeight : (LuceneConfig.WildcardsSearchNameWeight - boostOffset);
         break;
```
Please let me know what you think.

Reply:
I still don't get it, at the end query.Boost is assigned a value..so whatever value is assigned from the switch cae it will be overridden by the if block.

Reply:
@zeusongit yes, that is the purpose, the query.Boost value will be overridden only if (termSplit == true), in other words, will be executed only if the SearchTerm has empty space like "list join" and is split by empty space.

if you see it was already working in this way (see screnshot attached below).
https://github.com/DynamoDS/Dynamo/blob/eda5fb922210d2272f5dadb174b3cd2d6e34b5ec/src/DynamoCore/Utilities/LuceneSearchUtility.cs#L460C23-L467C26
![image](https://github.com/user-attachments/assets/c9fbbaf8-9bbe-4940-8738-94d075dcabd9)

Reply:
Yes, but my concern is, why not put this if..else block before the switch statement and exit if the condition is satisfied?

Reply:
Ok, I got your point I think I got confused, this change was applied in the next commit: https://github.com/DynamoDS/Dynamo/pull/15473/commits/31ad32ba6fa9bb5ecc0f66065c61605aafbd68ff